### PR TITLE
Added Metrics plugin

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-55218

### CODE REVIEW

File: source/scale/performance-monitoring.rst [hunk 1]
```diff
@@ -8,15 +8,39 @@ Performance monitoring
...
 
-   While Prometheus and Grafana may be installed on the same server as Mattermost, we recommend installing these integrations on separate servers, and configure Prometheus to pull all metrics from Mattermost and other connected servers.
+Mattermost Metrics plugin
+-------------------------
+
+The Mattermost Metrics plugin, available from Mattermost v6.3, collects and stores the `same performance monitoring metrics <#statistics>`__ as if Prometheus and Grafana were deployed, without having to deploy these third-party tools. Data is collected every minute and is stored where the plugin is running. The data is sychronized to either a cloud-based or local file store every hour, and retained for 15 days. 
+
+Using the Mattermost Metrics plugin, you can download and share the collected data with Mattermost to understand application performance, troubleshoot system stability and performance, as well as inform route cause analysis.
+
+In High Availability deployments, the Mattermost Metrics Plugin operates in 2 modes: scraper mode and listener mode.
```
Comment (cwarnermm) [inline source/scale/performance-monitoring.rst:22]: @isacikgoz - Can you provide details on how admins can operate the plugin in each mode, please?

File: source/scale/performance-monitoring.rst [hunk 2]
```diff
@@ -8,15 +8,39 @@ Performance monitoring
...
+- In **Scraper Mode**, the plugin scrapes metrics from nodes listed in the ``ClusterDiscovery`` table. The plugin includes Time Series Database (tsdb) synchronization to a remote file store. This mode is essential for collecting metrics data.
+
+- In **Listener Mode**, the plugin serves download requests by copying tsdb blocks from the remote file store and preparing the dump. This mode is focused on making the collected metrics available for analysis and troubleshooting.
+
+For the single node deployment, these modes are combined.
+
+Install the Mattermost Metrics plugin
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Download the latest version from the `Mattermost release page <https://github.com/mattermost/mattermost-plugin-metrics/releases>`__.
```
Comment (cwarnermm) [inline source/scale/performance-monitoring.rst:33]: @isacikgoz @nab-77 - When do we expect to have a release of this plugin on the release page? (it's currently empty)

File: source/scale/performance-monitoring.rst [hunk 3]
```diff
@@ -8,15 +8,39 @@ Performance monitoring
...
+Performance monitoring support enables admins to track system health for large Enterprise deployments through integrations with `Prometheus <https://prometheus.io/>`__ and `Grafana <https://grafana.org/>`__. These integrations support data collection from several Mattermost servers, which is particularly useful if you're running Mattermost `in high availability mode </scale/high-availability-cluster.html>`__. Once you're tracking system health, you can `set up performance alerts </scale/performance-alerting.html>`__ on your Grafana dashboard.
 
-.. note::
+Admins can collect and store various data points from the Mattermost application in an `OpenMetrics <https://openmetrics.io>`__ format by `deploying the Mattermost Metrics plugin <#mattermost-metrics-plugin>`__, or by `deploying Prometheus <#install-prometheus>`__ and `Grafana <#install-grafana>`__.
 
-   While Prometheus and Grafana may be installed on the same server as Mattermost, we recommend installing these integrations on separate servers, and configure Prometheus to pull all metrics from Mattermost and other connected servers.
+Mattermost Metrics plugin
```
Comment (cwarnermm) [inline source/scale/performance-monitoring.rst:15]: @nab-77 - This docs PR draft doesn't yet contain details on building your own dashboards. Can you or @isacikgoz provide draft details please?

File: source/scale/performance-monitoring.rst (hunk 1 above)
Comment (cwarnermm) [inline source/scale/performance-monitoring.rst:22]:
```suggestion
In High Availability deployments, the Mattermost Metrics plugin operates in 2 modes: scraper mode and listener mode.
```

Comment (isacikgoz) [reply source/scale/performance-monitoring.rst]: @cwarnermm that's an implementation detail actually, admins have no control on these modes so maybe we can emphasize that or remove this information entirely. Both works for me.

File: source/scale/performance-monitoring.rst (hunk 2 above)
Comment (isacikgoz) [reply source/scale/performance-monitoring.rst:33]: Once the security review completed, we will have an alpha version ready. https://mattermost.atlassian.net/browse/SEC-5160

File: source/scale/performance-monitoring.rst (hunk 3 above)
Comment (cwarnermm) [reply source/scale/performance-monitoring.rst:15]: @nab-77 ^ ?

File: source/scale/performance-monitoring.rst (hunk 1 above)
Comment (cwarnermm) [reply source/scale/performance-monitoring.rst]: If admins don't have control over the modes, best to remove from docs.

File: source/scale/performance-monitoring.rst (hunk 3 above)
Comment (isacikgoz) [reply source/scale/performance-monitoring.rst:15]: Discussed this with the team, yeah we can add that documentation here 👍

Comment (cwarnermm) [reply source/scale/performance-monitoring.rst:15]: Great! Thanks, @isacikgoz, for discussing with the team. Can you help draft initial content on building your own dashboards?

**Review by isacikgoz — APPROVED** [review]

**Comment by cwarnermm** [comment]
@nab-77 - Is there a release ETA for this new plugin we're working towards?

**Comment by nab-77** [comment]
> @nab-77 - Is there a release ETA for this new plugin we're working towards?

Currently running on community as the last check ahead of making available in "preview" mode.

**Comment by cwarnermm** [comment]
@nab-77 - Performance monitoring content is now split, as requested, into 3 parts: 3rd party integrations, our own plugin, and metrics. Looking forward to your feedback!

File: source/guides/scale-mattermost.rst [hunk 4]
```diff
@@ -18,6 +18,8 @@ Scale Mattermost
     High availability cluster </scale/high-availability-cluster>
     Elasticsearch </scale/elasticsearch>
     Performance monitoring </scale/performance-monitoring>
```
Comment (nab-77) [inline source/guides/scale-mattermost.rst:20]:
```suggestion
    Performance monitoring using Dashboards </scale/performance-monitoring>
```

File: source/guides/scale-mattermost.rst [hunk 5]
```diff
@@ -18,6 +18,8 @@ Scale Mattermost
...
+    Mattermost Metrics plugin </scale/metrics-plugin>
```
Comment (nab-77) [inline source/guides/scale-mattermost.rst:21]:
```suggestion
    Peformance monitoring using the plugin </scale/metrics-plugin>
```

**Review by nab-77 — CHANGES_REQUESTED** [review]
@cwarnermm one suggesting is to change the order of the items in the section:

1. Performance monitoring using the plugin
2. Performance monitoring using dashboards
3. Performance monitoring metrics

File: source/guides/scale-mattermost.rst (hunk 4 above)
Comment (cwarnermm) [reply source/guides/scale-mattermost.rst:20]: Incorporated into the latest commit.

File: source/guides/scale-mattermost.rst [hunk 6]
```diff
@@ -17,7 +17,9 @@ Scale Mattermost
     Scale up to 88000 users </scale/scale-to-88000-users>
     High availability cluster </scale/high-availability-cluster>
     Elasticsearch </scale/elasticsearch>
-    Performance monitoring </scale/performance-monitoring>
+    Monitor performance using the Metrics plugin </scale/metrics-plugin>
+    Monitor performance using dashboards </scale/performance-monitoring>
```
Comment (isacikgoz) [inline source/guides/scale-mattermost.rst:21]: Wondering if these two are sounding like alternative to each other. What about naming them like: 
```suggestion
    Collect performance metrics using the Metrics plugin </scale/metrics-plugin>
    Deploy Prometheus and Grafana for performance monitoring </scale/performance-monitoring>
```

File: source/guides/scale-mattermost.rst [hunk 7]
```diff
@@ -33,5 +35,7 @@ Scale and monitor your Mattermost deployment.
  * :doc:`Scale up to 88000 users </scale/scale-to-88000-users>` - Learn how to scale Mattermost to up to 88000 users.
 * :doc:`High availability cluster </scale/high-availability-cluster>` - Maintain Mattermost service during outages and hardware failures with redundant infrastructure.
 * :doc:`Elasticsearch </scale/elasticsearch>` - Enhance search performance with Elasticsearch.
-* :doc:`Performance monitoring </scale/performance-monitoring>` - Use Prometheus and Grafana to monitor the health and performance of your Mattermost cluster.
+* :doc:`Monitor performance using the Mattermost Metrics plugin </scale/metrics-plugin>` - Use the Mattermost Metrics Plugin for cases where Prometheus and Grafana aren't available.
```
Comment (isacikgoz) [inline source/guides/scale-mattermost.rst:38]:
```suggestion
* :doc:`Monitor performance using the Mattermost Metrics plugin </scale/metrics-plugin>` - Use the Mattermost Metrics Plugin for cases where Prometheus isn't available.
```

**Review by isacikgoz — COMMENTED** [review]
LGTM but I have some title suggestions.

**Comment by cwarnermm** [comment]
@isacikgoz - Really liked your titling suggestions! Thank you!

**Comment by cwarnermm** [comment]
An integration consolidation docs effort is in progress via https://github.com/mattermost/docs/pull/7030. My plan is to incorporate the the Metrics content from this PR there. Current ETA is by the end of the month (April).

**Pull request merged**